src/contexts/SwapProvider/SwapProvider.tsx

```diff
@@ -251,293 +215,405 @@ export function SwapContextProvider({ children }: { children: any }) {
-            },
-          ],
+          params: [swapTx],
         }),
       );
 
-      if (signError) {
-        throwError(signError);
+      if (signError || !swapTxHash) {
+        return throwError(signError);
       }
 
-      avaxProviderC.waitForTransaction(swapTxHash).then(async (tx) => {
-        const isSuccessful = tx && tx.status === 1;
-
-        captureEncrypted(isSuccessful ? 'SwapSuccessful' : 'SwapFailed', {
-          address: userAddress,
-          txHash: swapTxHash,
-          chainId: ChainId.AVALANCHE_MAINNET_ID,
-        });
-
-        const srcAsset = findSymbol(srcToken);
-        const destAsset = findSymbol(destToken);
-        const srcAssetAmount = new Big(srcAmount)
-          .div(10 ** srcDecimals)
-          .toString();
-        const destAssetAmount = new Big(destAmount)
-          .div(10 ** destDecimals)
-          .toString();
-
-        browser.notifications.create({
-          type: 'basic',
-          title: isSuccessful
-            ? t('Swap transaction succeeded! 🎉')
-            : t('Swap transaction failed! ❌'),
-          iconUrl: '../../../../images/icon-256.png',
-          priority: 2,
-          message: isSuccessful
-            ? t(
-                'Successfully swapped {{srcAmount}} {{srcToken}} to {{destAmount}} {{destToken}}',
-                {
-                  srcAmount: srcAssetAmount,
-                  destAmount: destAssetAmount,
-                  srcToken: srcAsset,
-                  destToken: destAsset,
-                },
-              )
-            : t(
-                'Could not swap {{srcAmount}} {{srcToken}} to {{destAmount}} {{destToken}}',
-                {
-                  srcToken: srcAsset,
-                  destToken: destAsset,
-                  srcAmount: srcAssetAmount,
-                  destAmount: destAssetAmount,
-                },
-              ),
-        });
+      notifyOnSwapResult({
+        provider: avaxProviderC,
+        txHash: swapTxHash,
+        userAddress,
+        srcToken,
+        destToken,
+        srcAmount,
+        destAmount,
+        srcDecimals,
+        destDecimals,
       });
-
-      return {
-        swapTxHash,
-        approveTxHash,
-      };
     },
     [
-      activeAccount?.addressC,
+      activeAccount,
       activeNetwork,
       avaxProviderC,
       buildTx,
-      captureEncrypted,
-      getParaswapSpender,
       networkFee,
       request,
-      t,
-      findSymbol,
+      notifyOnSwapResult,
+      getSwapTxProps,
     ],
   );
 
+  const swap = useCallback(
+    async (params: SwapParams) => {
+      const isOneClickSwapEnabled = isFlagEnabled(FeatureGates.ONE_CLICK_SWAP);
+      const isOneClickSwapSupported =
+        walletDetails?.type === SecretType.Mnemonic ||
+        walletDetails?.type === SecretType.Seedless ||
+        walletDetails?.type === SecretType.PrivateKey;
+
+      if (isOneClickSwapEnabled && isOneClickSwapSupported) {
+        return oneClickSwap(params);
+      }
```

Comment:
Please add tests for the new on-click swap path as well.

Reply:
Done!